boot/seal_test.go

```diff
@@ -526,8 +567,8 @@ func (s *sealSuite) TestResealKeyToModeenvWithSystemFallback(c *C) {
 		// verify the boot chains data file
 		pbc, cnt, err := boot.ReadBootChains(filepath.Join(dirs.SnapFDEDir, "boot-chains"))
 		c.Assert(err, IsNil)
-		if tc.prevPbc {
-			c.Assert(cnt, Equals, 10)
+		if tc.reuseRunPbc {
+			c.Assert(cnt, Equals, 9)
```

Comment:
why did this change from 10 -> 9 ?

Reply:
AFAIU the test was buggy here. In the case when sealedKeys == true and prevPbc == true, we'd hit a continue in line 510 in the old code. Right now it's visible that when a run/recovery mode predictable boot chain is reused, the respective count is not bumped.